vpr/src/base/vpr_api.cpp

```diff
@@ -383,6 +383,9 @@ bool vpr_flow(t_vpr_setup& vpr_setup, t_arch& arch) {
         if (!pack_success) {
             return false; //Unimplementable
         }
```

Comment:
Will this work if --route is specified?  In that case packing is skipped (read from file), placement and skipped (read from file) but routing is performed.  Same question is --analysis is specified.  This change looks like it could break those flows.

Reply:
doPlacement is an enum:

enum e_stage_action {
    STAGE_SKIP = 0,
    STAGE_LOAD,
    STAGE_DO,
    STAGE_AUTO
};

So the value checked on line 386 (!vpr_setup.PlacerOpts.doPlacement) is only true when doPlacement is set to STAGE_SKIP.  If Analysis or Routing is run but Placement is not, doPlacement is set to STAGE_LOAD, which does not trigger the early return in line 387.